test/integration/test_integration_basics.py

```diff
@@ -52,37 +55,60 @@ def mini_stress_test_h1(fixture, args):
   return counters
 
 
+# The mini stress tests below are executing in closed-loop mode. As we guard the pool against
+# overflows, we can set fixed expectations with respect to overflows and anticipated pending
+# totals.
 def test_http_h1_mini_stress_test_with_client_side_queueing(http_test_server_fixture):
   """
   Run a max rps test with the h1 pool against our test server, using a small client-side
-  queue. We expect to observe:
-  - upstream_rq_pending_total increasing
-  - upstream_cx_overflow overflows
-  - blocking to be reported by the sequencer
-  """
-  counters = mini_stress_test_h1(http_test_server_fixture, [
+  queue."""
+  counters = mini_stress_test(http_test_server_fixture, [
       http_test_server_fixture.getTestServerRootUri(), "--rps", "999999", "--max-pending-requests",
-      "10", "--duration 10"
+      "10", "--duration 10", "--connections", "1"
   ])
-  assertCounterGreater(counters, "upstream_rq_pending_total", 100)
-  assertCounterGreater(counters, "upstream_cx_overflow", 0)
+  assertCounterEqual(counters, "upstream_rq_pending_total", 11)
+  assertCounterEqual(counters, "upstream_cx_overflow", 10)
 
 
 def test_http_h1_mini_stress_test_without_client_side_queueing(http_test_server_fixture):
   """
   Run a max rps test with the h1 pool against our test server, with no client-side
-  queueing. We expect to observe:
-  - upstream_rq_pending_total to be equal to 1
-  - blocking to be reported by the sequencer
-  - no upstream_cx_overflows
+  queueing.
   """
-  counters = mini_stress_test_h1(
-      http_test_server_fixture,
-      [http_test_server_fixture.getTestServerRootUri(), "--rps", "999999", "--duration 2"])
+  counters = mini_stress_test(http_test_server_fixture, [
+      http_test_server_fixture.getTestServerRootUri(), "--rps", "999999", "--duration 2",
+      "--connections", "1"
+  ])
   assertCounterEqual(counters, "upstream_rq_pending_total", 1)
   assertNotIn("upstream_cx_overflow", counters)
 
 
+def test_http_h2_mini_stress_test_with_client_side_queueing(http_test_server_fixture):
+  """
+  Run a max rps test with the h2 pool against our test server, using a small client-side
+  queue. 
+  """
+  counters = mini_stress_test(http_test_server_fixture, [
+      http_test_server_fixture.getTestServerRootUri(), "--rps", "999999", "--max-pending-requests",
```

Comment:
Any chance 999999 is too small to truly stress it with this test harness? Is there any huge value like 999999999 that causes it to fail or behave differently? I'm remembering the time I got burned by using 9999 for infinity, and also the time Dr. Evil asked for one MILLION dollars :)

Reply:
Maybe we should consider adding a special value to indicate "as-fast-as-possible"? E.g. `MAX` or something like that. We would then internally translate to that to a large value.
For now I'd like to file an issue to track that, as this value is used in other places as well, and we update all relevant places in one go when Nighthawk understands that special value. WDYT?
(I think it is wishful thinking that mainstream processors will break this test anytime soon :-) )

Reply:
Yep, sounds good. There's also a value `requests_per_connection=1024 * 1024` in test/integration/test_connection_management.py that might fall under the same scope.

Reply:
Filed https://github.com/envoyproxy/nighthawk/issues/190